app/controllers/application_controller.rb

```diff
@@ -27,6 +29,19 @@ def redirect_when_logged_in
     end
   end
 
+  def require_valid_subscription
+    unless logged_in? && (current_user.valid_subscription? || current_user.admin?)
+      flash[:danger] = "Your subscription has been cancelled.  Please update your credit card details to resubscribe to myflix."
+      redirect_to account_details_path unless request.path == '/account_details'
+    end
+  end
+
+  def flash_delinquent_warning
+    if logged_in? && current_user.delinquent? && current_user.valid_subscription
+      flash[:warning] = "You recently had a failed payment.  In order to avoid interruptions in your Myflix service, please <a href='/account_details'>update your payment method</a>".html_safe
```

Comment:
Got it.  The reason I didn't initially use the named route was because `link_to` is not available in controller context.  I googled around and found out that I can just call `view_context.link_to...`